### ISSUE TITLE
IDE0250 should not report on structs containing ref struct fields

### ISSUE THREAD

**Issue by Corniel** [dotnet/roslyn#82960]
IDE0250 keeps track of structs that can be marked `readonly`. This leads to issues when any of its fields is a `ref struct`.
Take the following snippet:

``` C#
public ref struct Data(ReadOnlySpan<char> input, Span<char> buffer)
{
    private readonly ReadOnlySpan<char> Input = input;
    private readonly Span<char> Buffer = buffer;
}
```

By excluding structs that have `ref struct` properties, this False Positive will not longer be reported.

Previously, this rule has been adjusted to ignore structs containing events: #69994

**Comment by CyrusNajmabadi** [comment]
> This leads to issues when any of its fields is a ref struct.

What issues does this lead to?

**Comment by Corniel** [comment]
It suggest to a code change that does not compile. The error shown than states that the `struct` should be a `ref struct`.

**Comment by CyrusNajmabadi** [comment]
It compiles fine for me:

<img width="783" height="460" alt="Image" src="https://github.com/user-attachments/assets/cea0cf98-ac3f-4ffd-b229-43038d755f82" />

**Comment by Corniel** [comment]
Ah, You made it a `readonly ref struct`. That might be worth an improvement in the message, to make clear it should `readonly ref`.

**Comment by CyrusNajmabadi** [comment]
> That might be worth an improvement in the message,

We made the existing struct `readonly`.  We don't care what kind of struct it was :)

And the diff shows what we do:

<img width="897" height="486" alt="Image" src="https://github.com/user-attachments/assets/43c32f76-f446-42b0-9e9e-d3ebfa2d197c" />

:)

**Comment by RikkiGibson** [comment]
It sounds like lack of dotnet/csharplang#9804 is muddying the waters here. i.e. `ref readonly struct` is not valid, we have a proposal out to make it valid and mean the same thing as `readonly ref struct`.

I'll close this out as by design.
